### PR TITLE
Notifications rxjs

### PR DESCRIPTION
**Pull request by Wells-Codes**

### CODE REVIEW

File: core/src/js/components/notifications.component.ts [hunk 1]
```diff
@@ -77,7 +80,28 @@ export class NotificationsComponent implements AfterViewInit, OnDestroy {
...
 		private prefs: PreferencesService,
-	) {}
+	) {
+		this.notifications$ = this.ow.getMainWindow().notificationsEmitterBus;
+		this.windowId$ = from(this.ow.getCurrentWindow()).pipe(pluck('id'));
+
+		this.notifications$.pipe(
+			delayWhen(() => this.windowId$),
+			withLatestFrom(this.windowId$),
+			filter(([message, windowId]) => !!message),
```
Comment (Wells-Codes) [inline core/src/js/components/notifications.component.ts:90]: You asked why I used `!!` and it got me thinking. `filter` needs to return a boolean which message is not.

We could do `filter(message => { if(message) { return true; } else { return false; } })`
Which could be shortened to `filter(message => message ? true : false)`
Or even shorter `filter(message => !!message)`

Not that shorter is better, but that explains how I got in the habit of using `!!` with `filter`. Just wanted to follow up, cheers

Comment (sebastientromp) [reply core/src/js/components/notifications.component.ts:90]: Got it. When using `filter()` on arrays, it is enough to just do `filter(message => message)`, which removes null-ish items. I guess the signature is different then, and only accept real booleans, not true-ish values?

**Comment by sebastientromp** [comment]
Hey, I've cherry-picked the two commits into master (it lets me keeps a history that is more to my taste than merging the PRs, and the commits are left with your name :)).
I'll try to adapt this paradigm when I can moving forward, it might simplify some things quite a bit.

Thanks a lot for your help!

File: core/src/js/components/notifications.component.ts (hunk 1 above)
Comment (Wells-Codes) [reply core/src/js/components/notifications.component.ts:90]: Yeah, what's weird is that the build will mention an error, but it still seems to work anyway with just `filter(message => message)`

**Comment by Wells-Codes** [comment]
> Hey, I've cherry-picked the two commits into master (it lets me keeps a history that is more to my taste than merging the PRs, and the commits are left with your name :)).

No problem. I could always rebase and then you could fast-forward merge if you want to try that in the future. That should keep the history how you like

**Comment by sebastientromp** [comment]
If there's a way to ff merge the PR then that would be perfect (I always had the "merge branch" commit message when I tried to merge the PR, but haven't looked much into it).

**Pull request closed without merging**